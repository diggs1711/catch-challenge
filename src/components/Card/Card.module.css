.card {
    /* border: 1px solid #ccc; */
    max-width: 300px;
    margin: 1em 0.5em;
    border-radius: 0.5em;
    box-shadow: 1px 1px 15px 0.1px #ccc;
    cursor: pointer;
    transition: .2s all linear;
}

.card:hover {
    transform: scale(1.02);
}

.card__image {
    object-position: center;
    object-fit: cover;
}