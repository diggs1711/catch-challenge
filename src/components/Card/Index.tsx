import React from "react";
import styles from "../Card/Card.module.css";

type CardImageOptions = {
  url: string | undefined;
  alt: string;
};

type CardProps = {
  image?: CardImageOptions;
};

const Card: React.FC<CardProps> = ({ image, children }) => {
  return (
    <div className={styles.card}>
      {image ? (
        <img className={styles.card__image} src={image.url} alt={image.alt} />
      ) : null}
      <section>{children}</section>
    </div>
  );
};

export default Card;
