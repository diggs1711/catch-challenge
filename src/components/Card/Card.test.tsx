import React from "react";
import { render, screen } from "@testing-library/react";
import Card from "./Index";

describe("Card", () => {
  const testDetails = {
    imageUrl: "product_image.jpg",
    title: "Test title",
  };
  it("renders image and children", () => {
    render(
      <Card image={{ url: testDetails.imageUrl, alt: testDetails.title }}>
        <h6>{testDetails.title}</h6>
      </Card>
    );

    expect(screen.getByRole("img")).toBeInTheDocument();
    expect(screen.getByText(testDetails.title)).toBeInTheDocument();
  });

  it("renders when no image is supplied", () => {
    const { container } = render(
      <Card>
        <h6>{testDetails.title}</h6>
      </Card>
    );

    expect(container).toMatchSnapshot();
  });
});
