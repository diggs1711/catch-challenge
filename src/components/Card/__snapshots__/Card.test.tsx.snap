// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders when no image is supplied 1`] = `
<div>
  <div
    class="card"
  >
    <section>
      <h6>
        Test title
      </h6>
    </section>
  </div>
</div>
`;
